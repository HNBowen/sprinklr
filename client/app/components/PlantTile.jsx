import React from 'react'
import PropTypes from 'prop-types'

const PlantTile = (props) => {
  return (
      <div>
        <button onClick={function() { props.handleDelete(props.plant.id) }}>x</button>
        <img src={props.plant.image} onClick={function() { props.handleClick(props.plant.id) }}/>
        <span>{props.plant.name}</span>
        <span>Last watered: {props.plant.lastWatered}</span>
      </div>
    )
}

PlantTile.propTypes = {
  plant: PropTypes.object.isRequired,
  handleClick: PropTypes.func.isRequired
}

export default PlantTile