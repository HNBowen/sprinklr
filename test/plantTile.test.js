import React from 'react';
import {shallow} from 'enzyme'
import {shallowToJson} from 'enzyme-to-json'

import PlantTile from '../client/app/components/PlantTile.jsx'

describe('PlantTile', function() {

  it('should render correctly', function() {
    const wrapper = shallow(<PlantTile />)
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  })
})