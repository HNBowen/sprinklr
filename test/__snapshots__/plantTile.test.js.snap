// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlantTile should render correctly 1`] = `
<div>
  <img
    src="https://cdn.nurserylive.com/images/stories/virtuemart/product/resized/nurserylive-Peace-Lily,-Spathiphyllum-Plant_128x128.jpg"
  />
  <span>
    Peace Lily
  </span>
</div>
`;
